services/src/contexts/mod.rs

```diff
@@ -213,6 +216,40 @@ where
     ) -> Box<dyn InitializedPlotOperator> {
         op
     }
+
+    /// This method is meant to read a ml model from disk, specified by the config key `machinelearning.model_defs_path`.
+    async fn read_ml_model(
+        &self,
+        model_sub_path: std::path::PathBuf,
+    ) -> geoengine_operators::util::Result<String> {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidMachineLearningConfig)?;
+
+        let model_base_path = cfg.model_defs_path;
+
+        let model_path = canonicalize_subpath(&model_base_path, &model_sub_path)?;
+        let model = tokio::fs::read_to_string(model_path).await?;
+
+        Ok(model)
+    }
+
+    /// This method is meant to write a ml model to disk.
+    /// The provided path for the model has to exist.
+    async fn write_ml_model(
+        &mut self,
+        model_sub_path: std::path::PathBuf,
+        ml_model_str: String,
+    ) -> geoengine_operators::util::Result<()> {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidMachineLearningConfig)?;
+
+        let model_base_path = cfg.model_defs_path;
+
+        let model_path = canonicalize_subpath(&model_base_path, &model_sub_path)?;
```

Comment:
I think this does only work, if the path already exists. Please check this and use `path_with_base_path` instead if necessary.

Reply:
Exchanged the methods now ([see](https://github.com/geo-engine/geoengine/pull/639#discussion_r1034855518))